libs/datastructures/include/hpx/datastructures/tuple.hpp

```diff
@@ -54,7 +60,26 @@ namespace hpx { namespace util {
     HPX_CONSTEXPR HPX_HOST_DEVICE HPX_FORCEINLINE
         typename tuple_element<I, Tuple>::type const&&
         get(Tuple const&& t) noexcept;
+}}
 
+// Adapt hpx::util::tuple to be useable with std::tuple
+namespace std
+{
+    template <typename... Ts>
+    struct tuple_size<hpx::util::tuple<Ts...>>
+      : public hpx::util::tuple_size<hpx::util::tuple<Ts...>>
+    {};
+
+    template <std::size_t I, typename... Ts>
+    struct tuple_element<I, hpx::util::tuple<Ts...>>
+      : public hpx::util::tuple_element<I, hpx::util::tuple<Ts...>>
+    {};
+
+    using hpx::util::get;
```

Comment:
Isn't this a bit dangerous? This relies on the std library itself using `get<>()` in an unqualified way. Is that true for all std libraries we care about? Besides, doesn't this associate the HPX namespace with `namespace std`; could this have other implications?

At the very least I would suggest to only conditionally enable this integration, such that users would have to explicitly buy-in. 

I'm not sure if there are other ways to achieve this, but if there are, let's consider less dangerous techniques instead.

Reply:
The `using` was my interpretation of: http://irclog.cct.lsu.edu/ste~b~~b~ar/2019-07-06#239285;